File: Service/ClientCare/RMA.Service.ClientCare.Contracts/Entities/Member/RolePlayerPolicyTransaction.cs
using RMA.Service.Admin.MasterDataManager.Contracts.Enums;

using System.Collections.Generic;

namespace RMA.Service.ClientCare.Contracts.Entities.Member
{
    public class RolePlayerPolicyTransaction
    {
        public int RolePlayerPolicyTransactionId { get; set; } // RolePlayerPolicyTransactionId (Primary key)
        public int TenantId { get; set; } // TenantId
        public int RolePlayerId { get; set; } // RolePlayerId
        public int PolicyId { get; set; } // PolicyId
        public int CoverPeriod { get; set; } // CoverPeriod
        public TransactionTypeEnum TransactionType { get; set; } // TransactionTypeId
        public string DocumentNumber { get; set; } // DocumentNumber (length: 50)
        public decimal? TotalAmount { get; set; } // TotalAmount
        public System.DateTime EffectiveDate { get; set; } // EffectiveDate
        public System.DateTime? SentDate { get; set; } // SentDate
        public RolePlayerPolicyTransactionStatusEnum RolePlayerPolicyTransactionStatus { get; set; } // RolePlayerPolicyTransactionStatusId
        public bool IsDeleted { get; set; } // isDeleted
        public string CreatedBy { get; set; } // CreatedBy (length: 50)
        public System.DateTime CreatedDate { get; set; } // CreatedDate
        public string ModifiedBy { get; set; } // ModifiedBy (length: 50)
        public System.DateTime ModifiedDate { get; set; } // ModifiedDate
        public SourceProcessEnum? SourceProcess { get; set; }
        public System.DateTime? DocumentDate { get; set; } // DocumentDate

        public List<RolePlayerPolicyTransactionDetail> RolePlayerPolicyTransactionDetails { get; set; }
    }
}

File: Web/RMA-Web-Shell/projects/clientcare/src/app/policy-manager/shared/entities/role-player-policy-transaction.ts
import { TransactionTypeEnum } from "projects/fincare/src/app/shared/enum/transactionTypeEnum";
import { RolePlayerPolicyTransactionStatusEnum } from "../enums/role-player-policy-transaction-status.enum";
import { SourceProcessEnum } from "projects/shared-models-lib/src/lib/enums/source-enums";
import { RolePlayerPolicyTransactionDetail } from "./role-player-policy-transaction-detail";

export class RolePlayerPolicyTransaction {
  rolePlayerPolicyTransactionId: number;
  tenantId: number;
  rolePlayerId: number;
  policyId: number;
  coverPeriod: number;
  transactionType: TransactionTypeEnum;
  documentNumber: string;
  totalAmount: number;
  effectiveDate: Date;
  sentDate: Date;
  rolePlayerPolicyTransactionStatus: RolePlayerPolicyTransactionStatusEnum;
  sourceProcess: SourceProcessEnum;
  documentDate: Date;

  rolePlayerPolicyTransactionDetails: RolePlayerPolicyTransactionDetail[];
  
  isDeleted: boolean;
  createdBy: string;
  createdDate: Date;
  modifiedBy: string;
  modifiedDate: Date;
}

